nssp/delphi_nssp/pull.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+"""Functions for pulling NCHS mortality data API."""
+
+import numpy as np
+import pandas as pd
+from sodapy import Socrata
+
+from .constants import (
+    METRICS,
+    NEWLINE,
+)
+
+
+def construct_typedicts():
+    """Create the type conversion dictionary for dataframe."""
+    # basic type conversion
+    type_dict = {key: float for key in METRICS}
+    type_dict["timestamp"] = "datetime64[ns]"
+    type_dict["geography"] = str 
+    type_dict["county"] = str
+    type_dict["fips"] = int
+    return type_dict
+
+
+def warn_string(df, type_dict):
+    """Format the warning string."""
+    return f"""
+Expected column(s) missed, The dataset schema may
+have changed. Please investigate and amend the code.
+
+Columns needed:
+{NEWLINE.join(sorted(type_dict.keys()))}
+
+Columns available:
+{NEWLINE.join(sorted(df.columns))}
+"""
```

Comment:
nit: indent for readability (unless that messes up f-strings?)

Reply:
https://github.com/cmu-delphi/covidcast-indicators/pull/1952/commits/583e24eef7a6d7e75448976b89ab3d58e3e296dc